model/client4.go

```diff
@@ -1653,9 +1653,9 @@ func (c *Client4) UpdateChannelNotifyProps(channelId, userId string, props map[s
 }
 
 // AddChannelMember adds user to channel and return a channel member.
-func (c *Client4) AddChannelMember(channelId, userId string) (*ChannelMember, *Response) {
-	requestBody := map[string]string{"user_id": userId}
-	if r, err := c.DoApiPost(c.GetChannelMembersRoute(channelId)+"", MapToJson(requestBody)); err != nil {
+func (c *Client4) AddChannelMember(channelId, userId, postRootId string) (*ChannelMember, *Response) {
```

Comment:
If we're going with the server side solution, this needs to be made into a new client function `AddChannelMemberWithRootId` so as not to break backwards compatibility

Reply:
Noted, please see updated PR.